src/components/Character.cpp

```diff
@@ -1,28 +1,83 @@
 #include "Character.hpp"
+#include <Components/BsCCharacterController.h>
 #include <RTTI/RTTI_Character.hpp>
 #include <Scene/BsSceneObject.h>
+#include <components/CharacterAI.hpp>
+#include <components/CharacterEventQueue.hpp>
 #include <components/GameWorld.hpp>
 #include <components/VisualCharacter.hpp>
+#include <components/Waynet.hpp>
+#include <components/Waypoint.hpp>
 #include <scripting/ScriptVMForGameWorld.hpp>
 
 namespace REGoth
 {
-  Character::Character(const bs::HSceneObject& parent, const bs::String& instance)
-      : ScriptBackedBy(parent, "C_NPC", instance)
+  Character::Character(const bs::HSceneObject& parent, const bs::String& instance,
+                       HGameWorld gameWorld)
+    : ScriptBackedBy(parent, "C_NPC", instance, gameWorld)
   {
     setName("Character");
   }
 
   void Character::onInitialized()
   {
-    ScriptBackedBy::onInitialized();
+    // Only run this when we're not getting deserialized
+    if (!hasInstantiatedScriptObject())
```

Comment:
Since I've had to deal with similar checks, I wonder if there is a canonical bsf way of checking if we are being deserialized?

Reply:
I don't know. Haven't seen something like that yet, but I assume there must indeed be a better way.

Reply:
After checking some bsf-components, it seems like they also check whether some internal state is already there before initializing...